File: 0055-jump-game/0055-jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        
        last = len(nums) - 1
        
        for idx in range(len(nums) - 2, -1, -1):
            if idx + nums[idx] >= last:
                last = idx
            
        return last == 0
            
        

File: 0055-jump-game/0055-jump-game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {

        int last = nums.size() - 1;
        
        for(int i = nums.size() - 2; 0 <= i; i--) {
            if (i + nums[i] >= last) {
                last = i;
            }
        }
        return last == 0;
        
    }
};